platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrServerFactory.java

```diff
@@ -137,32 +169,77 @@ public static SolrServer getHttpSolrServer(String url, String coreName, String c
         String coreUrl = url + "/" + coreName;
         SolrServer server;
         try {
-            if (StringUtils.startsWith(url, "https")) {
-                CloseableHttpClient client = getHttpClient();
-                createSolrCore(url, coreName, configFile, client);
-                server = new HttpSolrServer(coreUrl, client);
-            } else {
-                createSolrCore(url, coreName, configFile, null);
-                server = new HttpSolrServer(coreUrl);
-            }
+            server = getSolrServer(url, coreName, configFile, coreUrl);
         } catch (SolrException ex) {
-            LOGGER.error("Unable to create HTTP Solr server client ({}): {}", coreUrl,
-                    ex.getMessage());
-            return null;
+            return pool.submit(new SolrServerFetcher(url, coreName, configFile, coreUrl));
         }
 
         LOGGER.info("Created HTTP Solr server client ({})", coreUrl);
+        return Futures.immediateFuture(server);
+    }
+
+    private static SolrServer getSolrServer(String url, String coreName, String configFile,
+            String coreUrl) throws SolrException {
+        SolrServer server;
+        if (StringUtils.startsWith(url, "https")) {
+            CloseableHttpClient client = getHttpClient();
+            createSolrCore(url, coreName, configFile, client);
+            server = new HttpSolrServer(coreUrl, client);
+        } else {
+            createSolrCore(url, coreName, configFile, null);
+            server = new HttpSolrServer(coreUrl);
+        }
         return server;
     }
 
     private static CloseableHttpClient getHttpClient() {
         SSLConnectionSocketFactory sslConnectionSocketFactory = new SSLConnectionSocketFactory(
-                getSslContext(), getProtocols(), getCipherSuites(),
+                getSslContext(),
+                getProtocols(),
+                getCipherSuites(),
                 SSLConnectionSocketFactory.BROWSER_COMPATIBLE_HOSTNAME_VERIFIER);
+        HttpRequestRetryHandler solrRetryHandler = new HttpRequestRetryHandler() {
+            private final Logger logger = LoggerFactory.getLogger(HttpRequestRetryHandler.class);
```

Comment:
generally I think we use all caps for final variables - i think `logger` should be `LOGGER`